demo.sh

```diff
@@ -439,7 +435,9 @@ transfer_create() {
 
     ## extract PSBT data
     local decoded_psbt
-    decoded_psbt="$(_trace "${BCLI[@]}" decodepsbt "$(base64 -w0 "$send_data/$PSBT")")"
+    # By default, linux wraps base64 output every 76 cols, so we use 'tr -d' to remove whitespaces.
+    # Note 'base64 -w0' doesn't work on Mac OS X, which has different flags.
+    decoded_psbt="$(_trace "${BCLI[@]}" decodepsbt "$(cat $send_data/$PSBT | base64 | tr -d '\r\n')")"
```

Comment:
please add `tr` to `check_tools`'s `required_tools` list

since this is duplicated, please move the base64 encoding and its comment to a dedicated function, e.g.:
```bash
base64_file_nowrap() {
    # By default, linux wraps base64 output every 76 cols, so we use 'tr -d' to remove whitespaces.
    # Note 'base64 -w0' doesn't work on Mac OS X, which has different flags.
    base64 "$1" | tr -d '\r\n'
}
```

then use it here and for PSBT signing below, e.g.
```bash
decoded_psbt="$(_trace "${BCLI[@]}" decodepsbt "$(base64_file_nowrap "$send_data/$PSBT")")"
```